mailbox/api/src/main/java/org/apache/james/mailbox/model/MailboxACL.java

```diff
@@ -19,235 +19,710 @@
+        anybody, authenticated, owner
+    }
 
-        /**
-         * Tells of what type is the name returned by {@link #getName()}.
-         * 
-         * @return type of the name returned by {@link #getName()}
-         */
-        NameType getNameType();
+    /**
+     * SETACL third argument prefix
+     */
+    public static final char ADD_RIGHTS_MARKER = '+';
 
-        /**
-         * If true the {@link MailboxACLRights} returned by
-         * {@link MailboxACLEntry#getRights()} should be interpreted as
-         * "negative rights" as described in RFC4314: If the identifier "-fred"
-         * is granted the "w" right, that indicates that the "w" right is to be
-         * removed from users matching the identifier "fred", even though the
-         * user "fred" might have the "w" right as a consequence of some other
-         * identifier in the ACL.
-         * 
-         * Note that {@link MailboxACLEntry#getName()} does not start with "-"
-         * when {@link MailboxACLEntry#getRights()} returns true.
-         * 
-         * @return
-         */
-        boolean isNegative();
+    /**
+     * Marks groups when (de)serializing {@link MailboxACLEntryKey}s.
+     *
+     * @see MailboxACLEntryKey#serialize()
+     */
+    public static final char DEFAULT_GROUP_MARKER = '$';
 
-        /**
-         * Returns a serialized form of this {@link MailboxACLEntryKey} as a
-         * {@link String}. Implementations should choose a consistent way how
-         * all of {@link #getName()}, {@link #getNameType()} and
-         * {@link #isNegative()} get serialized.
-         * 
-         * RFC4314 sction 2. states: All user name strings accepted by the LOGIN
-         * or AUTHENTICATE commands to authenticate to the IMAP server are
-         * reserved as identifiers for the corresponding users. Identifiers
-         * starting with a dash ("-") are reserved for "negative rights",
-         * described below. All other identifier strings are interpreted in an
-         * implementation-defined manner.
-         * 
-         * Dovecot and Cyrus mark groups with '$' prefix. See <a
-         * href="http://wiki2.dovecot.org/SharedMailboxes/Shared"
-         * >http://wiki2.dovecot.org/SharedMailboxes/Shared</a>:
-         * 
-         * <cite>The $group syntax is not a standard, but it is mentioned in RFC
-         * 4314 examples and is also understood by at least Cyrus IMAP. Having
-         * '-' before the identifier specifies negative rights.</cite>
-         * 
-         * @see MailboxACL#DEFAULT_GROUP_MARKER
-         * @see MailboxACL#DEFAULT_NEGATIVE_MARKER
-         * 
-         * @return serialized form as a {@link String}
-         */
-        String serialize();
-    }
+    /**
+     * Marks negative when (de)serializing {@link MailboxACLEntryKey}s.
+     *
+     * @see MailboxACLEntryKey#serialize()
+     */
+    public static final char DEFAULT_NEGATIVE_MARKER = '-';
+
+    /**
+     * SETACL third argument prefix
+     */
```

Comment:
I'm not sure this comment matches the line below

Reply:
It is! Please be kind, I'm not the comment author...